### PR TITLE
oncall: handle begining of month

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/repo/notify.py [hunk 1]
```diff
@@ -157,17 +157,27 @@ async def stalled_prs(self):
...
     async def oncall_string(self):
-        response = await self.session.get(CALENDAR)
+        now = datetime.datetime.now()
+        sunday = now - datetime.timedelta(days=now.weekday() + 1)
+        monday = now - datetime.timedelta(days=now.weekday())
+        priorweek = now - datetime.timedelta(14)
+
+        # Handle the event being created before today.
+        date = (
+            "%s%s%s" % (priorweek.year, str(priorweek.month).zfill(2), str(priorweek.day).zfill(2)))
```
Comment (phlax) [inline tools/repo/notify.py:167]:
```suggestion
           priorweek.strftime("%Y%m%d")
```

File: tools/repo/notify.py [hunk 2]
```diff
@@ -157,17 +157,27 @@ async def stalled_prs(self):
...
+        response = await self.session.get(CALENDAR + "?getdate=" + date)
```
Comment (phlax) [inline tools/repo/notify.py:168]:
```suggestion
        response = await self.session.get(f"{CALENDAR}?getdate={date}")
```

prefer f strings

File: tools/repo/notify.py [hunk 3]
```diff
@@ -157,17 +157,27 @@ async def stalled_prs(self):
...
         content = await response.read()
         parsed_calendar = icalendar.Calendar.from_ical(content)
 
         now = datetime.datetime.now()
-        sunday = now - datetime.timedelta(days=now.weekday() + 1)
```
Comment (phlax) [inline tools/repo/notify.py:165]: i guess the `now` above this line can also go

**Review by phlax — COMMENTED** [review]
apart from nits lgtm, but not entirely sure i follow the fix or why its necessary

**Comment by alyssawilk** [comment]
so apparently if you just get basic.ics you get events today and after.  given that today's oncall event is a week long event starting last week, it doesn't show up when you download basic.ics, so you need basic.ics?lastweek (syntax in the script)

I don't know why the event then starts on a different day, will have to see how that plays out tomorrow :-/

**Comment by alyssawilk** [comment]
/retest
@phlax anything else you want to see here?

**Comment by alyssawilk** [comment]
(FWIW I checked this morning and still works, where checked in script is pinging me still)

**Review by phlax — APPROVED** [review]
lgtm, thanks @alyssawilk

**Pull request merged**